src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+    await flatTablePageSelectPrevious(page).locator("button").focus();
+    await page.keyboard.press("Space");
+    await expect(currentPageInput).toHaveAttribute("value", "1");
+    await expect(flatTableBodyRowByPosition(page, 4)).toBeInViewport();
+  });
+
+  test(`should navigate to previous page by clicking Previous link with the Enter key`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTablePagerStickyHeaderComponent />);
+
+    await flatTablePageSelectNext(page).click();
+    await expect(flatTableBodyRowByPosition(page, 0)).toBeInViewport();
+    const currentPageInput = flatTableCurrentPageInput(page);
+    await expect(currentPageInput).toHaveAttribute("value", "2");
+    await flatTablePageSelectPrevious(page).locator("button").focus();
+    await page.keyboard.press("Enter");
+    await expect(currentPageInput).toHaveAttribute("value", "1");
+
+    for (let i = 0; i < 5; i++) {
+      if (i < 4) {
+        await expect(flatTableBodyRowByPosition(page, i)).toBeInViewport();
+      } else {
+        await expect(flatTableBodyRowByPosition(page, i)).not.toBeInViewport();
+      }
+    }
+  });
+
+  test(`should navigate to next page by page number`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTablePagerStickyHeaderComponent />);
+
+    await expect(flatTableCurrentPageInput(page)).toHaveAttribute("value", "1");
+
+    for (let i = 0; i < 5; i++) {
+      if (i < 4) {
+        await expect(flatTableBodyRowByPosition(page, i)).toBeInViewport();
+      } else {
+        await expect(flatTableBodyRowByPosition(page, i)).not.toBeInViewport();
+      }
+    }
+
+    const currentPageInput = flatTableCurrentPageInput(page);
+    await currentPageInput.focus();
+    await page.keyboard.press("Backspace");
+    await page.keyboard.type("2");
+    await page.keyboard.press("Tab");
+    await expect(currentPageInput).toHaveAttribute("value", "2");
+    await expect(flatTableBodyRowByPosition(page, 0)).toBeInViewport();
+  });
+
+  test(`should navigate to previous page by page number`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTablePagerStickyHeaderComponent />);
+
+    await flatTablePageSelectNext(page).click();
+    await expect(flatTableBodyRowByPosition(page, 0)).toBeInViewport();
+    const currentPageInput = flatTableCurrentPageInput(page);
+    await expect(currentPageInput).toHaveAttribute("value", "2");
+    await currentPageInput.focus();
+    await page.keyboard.press("Backspace");
+    await page.keyboard.type("1");
+    await page.keyboard.press("Tab");
+    await expect(currentPageInput).toHaveAttribute("value", "1");
+
+    for (let i = 0; i < 5; i++) {
+      if (i < 4) {
+        await expect(flatTableBodyRowByPosition(page, i)).toBeInViewport();
+      } else {
+        await expect(flatTableBodyRowByPosition(page, i)).not.toBeInViewport();
+      }
+    }
+  });
+
+  test(`should render with clickable rows when focusRedesignOptOut is true`, async ({
```

Comment:
Should this specify "using mouse" to distinguish it from all the up/down arrow keys clickable tests above?